app/jobs/concerns/session_advisory_lock.rb

```diff
@@ -0,0 +1,36 @@
+require 'zlib'
+
+module SessionAdvisoryLock
+  extend ActiveSupport::Concern
+
+  class LockFailedError < RuntimeError; end
+
+  included do
+    delegate :connection, to: :"ActiveRecord::Base"
+    delegate :select_value, to: :connection
+
+    lock_id = Zlib.crc32(self.name)
```

Comment:
Advisory locks have to be a 64 bit integer so we just use a crc32 of the job name